### PR TITLE
mongodb

### LINKED ISSUE #23: mongodb

### PR DESCRIPTION
**Pull request by eyakauleva**
close #23

### CODE REVIEW

File: src/main/java/com/solvd/micro9/users/service/SequenceGeneratorService.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.solvd.micro9.users.service;
+
+import java.util.concurrent.ExecutionException;
+
+public interface SequenceGeneratorService {
+
+    Long generateSequence(final String sequenceName) throws InterruptedException, ExecutionException;
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/users/service/SequenceGeneratorService.java:7]: @eyakauleva, JFYI: This logic should be represented on MongoDB side: https://www.mongodb.com/basics/mongodb-auto-increment

File: src/main/java/com/solvd/micro9/users/service/impl/SequenceGeneratorServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@Service
+@RequiredArgsConstructor
+public class SequenceGeneratorServiceImpl implements SequenceGeneratorService {
+
+    private final ReactiveMongoOperations mongoOperations;
+
+    @Override
+    public Long generateSequence(final String sequenceName) throws InterruptedException, ExecutionException {
+        return mongoOperations.findAndModify(new Query(Criteria.where("_id").is(sequenceName)),
+                        new Update().inc("sequence", 1), options().returnNew(true).upsert(true), DatabaseSequence.class)
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/users/service/impl/SequenceGeneratorServiceImpl.java:25]: the line is too long. 80 indent is an absolutely maximum

File: src/main/java/com/solvd/micro9/users/web/controller/exception/ExceptionHandling.java [hunk 3]
```diff
@@ -2,14 +2,32 @@
...
 
+    @ExceptionHandler(WebExchangeBindException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public Mono<ExceptionBody> handleException(WebExchangeBindException ex) {
+        List<BindingError> bindingErrors = ex.getBindingResult()
+                .getAllErrors()
+                .stream()
+                .map(fieldError ->
+                        new BindingError(
+                                fieldError.getObjectName() + "." + ((FieldError) fieldError).getField(), fieldError.getDefaultMessage()))
```
Comment (h1alexbel) [inline src/main/java/com/solvd/micro9/users/web/controller/exception/ExceptionHandling.java:26]: line is too long

**Review by h1alexbel — COMMENTED** [review]
@eyakauleva, please fix the lines

File: src/main/java/com/solvd/micro9/users/service/impl/SequenceGeneratorServiceImpl.java (hunk 2 above)
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/users/service/impl/SequenceGeneratorServiceImpl.java]: fixed 88824df

File: src/main/java/com/solvd/micro9/users/web/controller/exception/ExceptionHandling.java (hunk 3 above)
Comment (eyakauleva) [reply src/main/java/com/solvd/micro9/users/web/controller/exception/ExceptionHandling.java]: fixed 88824df

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**